Runtime/codebase/PhantomWallet/PhantomDeepLink.cs

```diff
@@ -0,0 +1,174 @@
+                phantomConnectionAccountPrivateKey: PhantomConnectionAccountPrivateKey,
+                redirectScheme:  _phantomWalletOptions.deeplinkUrlScheme,
+                apiVersion: _phantomWalletOptions.phantomApiVersion,
+                sessionId: _sessionId,
+                cluster: RpcCluster
+                
+            );
+            Application.OpenURL(url);
+        }
+        
+        #endregion        
+
+        #region Callbacks
+        
+        private void OnDeepLinkActivated(string url)
+        {
+            if (url.Contains("transactionSigned"))
+            {
+                ParseSuccessfullySignedTransaction(url);
+            }
+            else if(url.Contains("onPhantomConnected"))
+            {
+                ParseConnectionSuccessful(url);
+            }
+            else
+            {
+                throw new NotImplementedException($"Callback not implemented for: {url}");
+            }
+        }
+
+        private void ParseConnectionSuccessful(string url)
+        {
+            var phantomResponse = url.Split("?"[0])[1];
+            var result = HttpUtility.ParseQueryString(phantomResponse);
+            _phantomEncryptionPubKey = Encoders.Base58.DecodeData(result.Get("phantom_encryption_public_key"));
+            var phantomNonce = result.Get("nonce");
+            var data = result.Get("data");
+            var errorMessage = result.Get("errorMessage");
+            if (!string.IsNullOrEmpty(errorMessage))
+            {
+                Debug.LogError($"Deeplink error: {errorMessage}");
+                _loginTaskCompletionSource.SetResult(null);
+                return;
+            }
+            if (string.IsNullOrEmpty(data))
+            {
+                Debug.LogError("Phantom connect canceled.");
+                _loginTaskCompletionSource.SetResult(null);
+                return;
+            }
+            var k = MontgomeryCurve25519.KeyExchange(_phantomEncryptionPubKey, PhantomConnectionAccountPrivateKey);
+            var unencryptedMessage = XSalsa20Poly1305.TryDecrypt(
+                Encoders.Base58.DecodeData(data), k, Encoders.Base58.DecodeData(phantomNonce));
+            var bytesToUtf8String = Encoding.UTF8.GetString(unencryptedMessage);
+            var connectSuccess = JsonUtility.FromJson<PhantomWalletConnectSuccess>(bytesToUtf8String);
+            var error = JsonUtility.FromJson<PhantomWalletError>(bytesToUtf8String);
+            if (!string.IsNullOrEmpty(connectSuccess.public_key))
+            {
+                _sessionId = connectSuccess.session;
+                _loginTaskCompletionSource.SetResult(new Account(string.Empty, connectSuccess.public_key));
+            }
+            else
+            {
+                if (string.IsNullOrEmpty(error.errorCode)) return;
+                _loginTaskCompletionSource.SetResult(null);
+                Debug.LogError($"Deeplink error: {error.errorCode} {error.errorMessage}");
```

Comment:
we should add some kind of logger at some point

Reply:
Agree, let's open a new issue for this